### PR TITLE
PERF/UCP: IOV support in PERF UCP part

### PR DESCRIPTION
**Pull request by shssf**
Implemented IOV list options in UCP part of the test
This is related to #951

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3139/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/689/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2637/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3140/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/690/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/693/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2638/ for details (Mellanox internal link).

File: src/tools/perf/libperf.c [hunk 1]
```diff
@@ -766,8 +766,24 @@ static ucs_status_t ucp_perf_test_alloc_mem(ucx_perf_context_t *perf, ucx_perf_p
     }
     perf->recv_buffer = mem_map_params.address;
 
+    /* Allocate IOV datatype memory */
+    if(UCP_PERF_DATATYPE_IOV == params->ucp.datatype) {
```
Comment (yosefe) [inline src/tools/perf/libperf.c]: space after if

File: src/tools/perf/ucp_tests.cc [hunk 2]
```diff
@@ -31,6 +31,37 @@ class ucp_perf_test_runner {
...
+
+                if (m_perf.params.iov_stride) {
+                    iov_length_it += m_perf.params.iov_stride;
+                } else {
+                    iov_length_it += m_perf.ucp.iov[iov_it].length;
+                }
+            }
+
+            ucs_debug("IOV buffer filled by %lu slices with total length %lu",
+                    iovcnt, iov_length_it);
```
Comment (yosefe) [inline src/tools/perf/ucp_tests.cc]: indentation

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3157/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3163/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3164/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3165/ for details.

**Comment by shamisp** [comment]
besides highlighted styling issues, it looks good to me.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3166/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3169/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/700/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2646/ for details (Mellanox internal link).

**Pull request merged**